### PR TITLE
FEAT: add dialog manager

### PR DESCRIPTION
**Pull request by theodorusclarence**
## Description

Dialog manager for easy confirmation 😎

Usage:
https://github.com/theodorusclarence/tanyaaja/blob/7bfe71809251da54f29fe2b522d790e52891889d/src/app/account/settings/page.tsx#L327-L342

btw i'm not really familiar with the folder placement, lmk if you want to move them

### CODE REVIEW

**Comment by theodorusclarence** [comment]
Currently considering to add loading indicator on the submit button, but still thinking on how to implement that.

in my projects, i [usually listens from react-hot-toast ](https://github.com/theodorusclarence/aether-design-system/blob/main/src/components/dialog/BaseDialog.tsx#L167)

since this project doesn't use a lot of loading toast, it can't be implemented.

I'll open a separate PR for the loading

File: package.json [hunk 1]
```diff
@@ -41,6 +41,7 @@
     "eslint": "8.51.0",
     "eslint-config-next": "13.5.4",
     "firebase": "^10.4.0",
+    "immer": "^10.0.3",
```
Comment (mazipan) [inline package.json]: Personally I don't convenient with how immer works. I think it adding unnecessary complexity in the code.

I don't think it is good to introduce this in the current stage.

Comment (mazipan) [reply package.json]: How about we remove this one?

Comment (theodorusclarence) [reply package.json]: immer is usually used with zustand to make the code more readable

sure, let me try to refactor them

Comment (mazipan) [reply package.json]: I already used zustand in many projects also, but without immer.

And I can say, it still works like charm.

So based on my experiences, it's just not needed in my workflow.

Of course, this is just personal preference.

**Review by mazipan — APPROVED** [review]
Looks good 👍

**Pull request merged**